pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/files/ListFiles.ios.kt

```diff
@@ -1,9 +1,54 @@
 package com.pubnub.api.endpoints.files
 
+import cocoapods.PubNubSwift.PubNubFileObjC
+import cocoapods.PubNubSwift.PubNubObjC
+import cocoapods.PubNubSwift.listFilesWithChannel
 import com.pubnub.api.Endpoint
 import com.pubnub.api.models.consumer.files.PNListFilesResult
+import com.pubnub.api.models.consumer.files.PNUploadedFile
+import com.pubnub.api.models.consumer.objects.PNPage
+import com.pubnub.api.onFailureHandler
+import com.pubnub.api.onSuccessHandler2
+import com.pubnub.api.v2.callbacks.Consumer
+import com.pubnub.api.v2.callbacks.Result
+import com.pubnub.kmp.createPubNubHashedPage
+import com.pubnub.kmp.filteredList
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.Foundation.NSNumber
 
 /**
  * @see [PubNub.listFiles]
  */
-actual interface ListFiles : Endpoint<PNListFilesResult>
\ No newline at end of file
+actual interface ListFiles : Endpoint<PNListFilesResult>
+
+@OptIn(ExperimentalForeignApi::class)
+class ListFilesImpl(
+    private val pubnub: PubNubObjC,
+    private val channel: String,
+    private val limit: Int?,
+    private val next: PNPage.PNNext?
+): ListFiles {
+    override fun async(callback: Consumer<Result<PNListFilesResult>>) {
+        pubnub.listFilesWithChannel(
+            channel = channel,
+            limit = limit?.let { NSNumber(it) },
+            next = createPubNubHashedPage(from = next),
+            onSuccess = callback.onSuccessHandler2 { data, next ->
+                PNListFilesResult(
+                    count = data?.size ?: 0, // TODO: count property is not retrieved from ListFilesSuccessResponse in Swift SDK
+                    next = next?.let { PNPage.PNNext(pageHash = it) },
+                    status = 200,
+                    data = filteredList(from = data) { rawValue: PubNubFileObjC ->
```

Comment:
For me it is a bit misleading that `data` is used for two purpose here. First as lambda param and second as property of PNListFilesResult. What about 
`onSuccess = callback.onSuccessHandler2 { input, nextInput->`
?

Reply:
Sure, I'll change parameter names from the lambda expression:

```
onSuccess = callback.onSuccessHandler2 { files, nextPageHash
```